contrib/stack/topsStack/fetchOrbit.py

```diff
@@ -146,28 +146,47 @@ def fileToRange(fname):
     for spec in orbitMap:
         oType = spec[0]
 
-        url = server + spec[1] + str(fileTS.year).zfill(2) + '/' + str(fileTS.month).zfill(2) + \
-                               '/' + str(fileTS.day).zfill(2) + '/'
+        if oType == 'precise':
+            end_date = fileTS + datetime.timedelta(days=20)
+        elif oType == 'restituted':
+            end_date = fileTS
+        else:
+            raise ValueError("Unexpected orbit type: '" + oType + "'")
+        end_date2 = end_date + datetime.timedelta(days=1)
+        url = server + spec[1] + str(end_date.year).zfill(2) + '/' + str(end_date.month).zfill(2) + \
+            '/' + str(end_date.day).zfill(2) + '/', server + spec[1] + str(end_date2.year).zfill(2) + '/' + str(end_date2.month).zfill(2) + \
+            '/' + str(end_date2.day).zfill(2) + '/'
```

Comment:
```suggestion
        url = (server + spec[1] + end_date.strftime("%Y/%m/%d/") for end_date in (end_date, end_date2))
```
Might be simpler with the builtin datetime formatting. Also, now that this is an iterable, I'd call it `urls` so that you can say `for url in urls` below.